contracts/FlowToken.cdc

```diff
@@ -73,36 +106,52 @@ pub contract FlowToken: FungibleToken, ViewResolver {
         // It is allowed to destroy the sent Vault because the Vault
         // was a temporary holder of the tokens. The Vault's balance has
         // been consumed and therefore can be destroyed.
-        pub fun deposit(from: @FungibleToken.Vault) {
+        access(all) fun deposit(from: @{FungibleToken.Vault}) {
             let vault <- from as! @FlowToken.Vault
             self.balance = self.balance + vault.balance
-            emit TokensDeposited(amount: vault.balance, to: self.owner?.address)
-            vault.balance = 0.0
-            destroy vault
-        }
 
-        destroy() {
-            if self.balance > 0.0 {
-                FlowToken.totalSupply = FlowToken.totalSupply - self.balance
+            // If the owner is the staking account, do not emit the contract defined events
+            // this is to help with the performance of the epoch transition operations
+            // Either way, event listeners should be paying attention to the 
+            // FungibleToken.Deposited events anyway because those contain
+            // much more comprehensive metadata
+            // Additionally, these events will eventually be removed from this contract completely
+            // in favor of the FungibleToken events
```

Comment:
I need to double check with Vishal about that, but we had decided that we didn't want to remove it as part of Crescendo because there were partners who it would be a breaking change for. I still would prefer to remove it since there are other breaking changes. I'll ask again. I created an issue to track it: https://github.com/onflow/flow-core-contracts/issues/434